src/commands/ssh-resolve.ts

```diff
@@ -0,0 +1,119 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { authenticate } from "../drivers/auth";
+import { bootstrapConfig } from "../drivers/env";
+import { fsShutdownGuard } from "../drivers/firestore";
+import { conditionalAbortBeforeThrow, P0_PATH } from "../util";
+import {
+  SSH_PROVIDERS,
+  SshResolveCommandArgs,
+  prepareRequest,
+} from "./shared/ssh";
+import fs from "fs";
+import path from "path";
+import tmp from "tmp-promise";
+import yargs from "yargs";
+
+export const sshResolveCommand = (yargs: yargs.Argv) =>
+  yargs.command<SshResolveCommandArgs>(
+    "ssh-resolve <destination>",
+    "SSH into a virtual machine",
+    (yargs) =>
+      yargs
+        .positional("destination", {
+          type: "string",
+          demandOption: true,
+        })
+        .option("parent", {
+          type: "string",
+          describe:
+            "The containing parent resource which the instance belongs to (account, project, subscription, etc.)",
+        })
+        .option("provider", {
+          type: "string",
+          describe: "The cloud provider where the instance is hosted",
+          choices: ["aws", "azure", "gcloud"],
+        })
+        .option("debug", {
+          type: "boolean",
+          describe: "Print debug information.",
+        })
+        .option("quiet", {
+          alias: "q",
+          type: "boolean",
+          describe: "Suppress output",
+        }),
+
+    fsShutdownGuard(sshResolveAction)
+  );
+
+/** Connect to an SSH backend
+ *
+ * Implicitly gains access to the SSH resource if required.
+ *
+ * Supported SSH mechanisms:
+ * - AWS EC2 via SSM with Okta SAML
+ */
+const sshResolveAction = async (
+  args: yargs.ArgumentsCamelCase<SshResolveCommandArgs>
+) => {
+  const p0Executable = bootstrapConfig.appPath;
+  // Prefix is required because the backend uses it to determine that this is an AWS request
```

Comment:
this comment also doesn't make any sense to me, i don't see prefix being used here.